README.md

```diff
@@ -1,94 +1,187 @@
+
+        Change this example to suit your needs. You'll need to specify the right entrypoint and
+        environment configuration for your specific workload.
+
+        Learn more about interacting with Pebble at at https://juju.is/docs/sdk/pebble.
+        """
+        # Get a reference the container attribute on the PebbleReadyEvent
+        container = event.workload
+        # Add initial Pebble config layer using the Pebble API
+        container.add_layer("httpbin", self._pebble_layer, combine=True)
+        # Make Pebble reevaluate its plan, ensuring any services are started if enabled.
+        container.replan()
+        # Learn more about statuses in the SDK docs:
+        # https://juju.is/docs/sdk/constructs#heading--statuses
+        self.unit.status = ops.ActiveStatus()
+...
+```
 
-The framework provides a testing harness, so you can ensure that your charm
-does the right thing in different scenarios, without having to create
-a full deployment. Our [API documentation](https://ops.readthedocs.io/en/latest/#module-ops.testing)
-has the details, including this example:
+> See more: [`ops.PebbleReadyEvent`](https://ops.readthedocs.io/en/latest/index.html#ops.PebbleReadyEvent)
 
-```python
+Note that the `tests/test_charm.py` file imports `ops.testing` and uses it to set up a testing harness:
+
+```
+...
+import ops.testing
+...
 class TestCharm(unittest.TestCase):
-    def test_foo(self):
-        harness = Harness(MyCharm)
-        self.addCleanup(harness.cleanup)  # always clean up after ourselves
+    def setUp(self):
+        self.harness = ops.testing.Harness(MyNewCharmCharm)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+...
+```
 
-        # Instantiate the charm and trigger events that Juju would on startup
-        harness.begin_with_initial_hooks()
+> See more: [`ops.testing.Harness`](https://ops.readthedocs.io/en/latest/#ops.testing.Harness)
 
-        # Update charm config and trigger config-changed
-        harness.update_config({'log_level': 'warn'})
 
-        # Check that charm properly handled config-changed, for example,
-        # the charm added the correct Pebble layer
-        plan = harness.get_container_pebble_plan('prometheus')
-        self.assertIn('--log.level=warn', plan.services['prometheus'].command)
+Explore further, start editing the files, or skip ahead and pack charm:
+
+```
+charmcraft pack
 ```
 
+If you didn't take any wrong turn or simply left the charm exactly as it was, this should work and yield a file called `my-new-charm_ubuntu-22.04-amd64.charm` (the architecture bit may be different depending on your system's architecture). Use this name and the resource from the `metadata.yaml` to deploy your example charm to your local MicroK8s cloud:
+
+```
+juju deploy ./my-new-charm_ubuntu-22.04-amd64.charm --resource httpbin-image=kennethreitz/httpbin
+```
+
+Congratulations, you’ve just built your first Kubernetes charm using Ops!
+
+### Clean up
```

Comment:
If we decide to remove the "Set up" section and just link it, we can do the same here.

Reply:
Updated the dev-setup such that we can easily link to the cleanup bit: https://juju.is/docs/sdk/dev-setup#heading--automatic-clean-up